controller/Lists.php

```diff
@@ -391,82 +285,83 @@ public function saveLists(ValueCollectionService $valueCollectionService): void
 
         $elements = $valueCollectionService->findAll(
             new ValueCollectionSearchInput(
-                (new ValueCollectionSearchRequest())
-                    ->setValueCollectionUri($listClass->getUri())
+                (new ValueCollectionSearchRequest())->setValueCollectionUri($listClass->getUri())
             )
         );
 
         foreach ($payload as $key => $value) {
             if (preg_match('/^list-element_/', $key)) {
                 $encodedUri = preg_replace('/^list-element_[0-9]+_/', '', $key);
-                $uri        = tao_helpers_Uri::decode($encodedUri);
-
+                $uri = tao_helpers_Uri::decode($encodedUri);
                 $newUriValue = trim($payload["uri_$key"] ?? '');
-
                 $element = $elements->extractValueByUri($uri);
 
-                if (null === $element) {
+                if ($element === null) {
                     $elements->addValue(new Value(null, $newUriValue, $value));
-                } else {
-                    $element->setLabel($value);
 
-                    if ($newUriValue) {
-                        $element->setUri($newUriValue);
-                    }
+                    continue;
                 }
 
+                $element->setLabel($value);
+
+                if ($newUriValue) {
+                    $element->setUri($newUriValue);
+                }
             }
         }
 
         try {
             $this->returnJson(['saved' => $valueCollectionService->persist($elements)]);
         } catch (ValueConflictException $exception) {
-            $this->returnJson(['saved' => false, 'errors' => [__('The list should contain unique URIs')]]);
+            $this->returnJson(
+                [
+                    'saved' => false,
+                    'errors' => [
+                        __('The list should contain unique URIs'),
+                    ],
+                ]
+            );
         }
     }
 
     /**
-     * Create a list or a list element
-     * @throws common_exception_BadRequest
-     * @return void
+     * @throws common_Exception
+     * @throws core_kernel_persistence_Exception
      */
-    public function create()
+    public function create(): void
     {
         if (!$this->isXmlHttpRequest()) {
             throw new common_exception_BadRequest('wrong request mode');
         }
 
         $response = [];
-        if ($this->getRequestParameter('classUri')) {
-            if ($this->getRequestParameter('type') === 'class' && $this->getRequestParameter('classUri') ==='root') {
-                $listClass = $this->getListService()->createList();
-                if (!is_null($listClass)) {
-                    $response['label']  = $listClass->getLabel();
-                    $response['uri']    = tao_helpers_Uri::encode($listClass->getUri());
+
+        if ($this->hasRequestParameter('classUri')) {
```

Comment:
We could use the `ListCreator` class here instead of having all this logic in the controller. Example: 

```
ListCreator::createByRequest(ServerRequest $request): ListCreatedResponse
```

Reply:
Totally agree with you.
This is one of my tasks for this PR (if I have time for this)

Thanks! 🙂